src/icp_v2.cc

```diff
@@ -427,7 +427,8 @@ icpCreateAndSend(icp_opcode opcode, int flags, char const *url, int reqnum, int
 void
 icpDenyAccess(Ip::Address &from, char *url, int reqnum, int fd)
 {
-    debugs(12, 2, "icpDenyAccess: Access Denied for " << from << " by " << AclMatchedName << ".");
```

Comment:
To preserve this debugging information, move this debugs() into icpAccessAllowed(), making it conditional on the future `false` result.

Reply:
Done.